File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/typeParameter/InheritMutability.java
package test;

import org.jetbrains.annotations.NotNull;
import jet.runtime.typeinfo.KotlinSignature;
import org.jetbrains.kotlin.jvm.compiler.annotation.ExpectLoadError;
import java.util.*;
import java.util.List;

public interface InheritMutability {

    public interface Super {
        @KotlinSignature("fun <A: MutableList<String>> foo(a: A)")
        <A extends List<String>> void foo(A a);
    }

    public interface Sub extends Super {
        <B extends List<String>> void foo(B b);
    }
}

File: compiler/testData/loadJava/compiledJavaCompareWithKotlin/kotlinSignature/propagation/parameter/InheritMutability.kt
package test

public trait InheritMutability: Object {

    public trait Super: Object {
        public fun foo(p0: MutableList<String>)

        public fun dummy() // to avoid loading as SAM interface
    }

    public trait Sub: Super {
        override fun foo(p0: MutableList<String>)
    }
}
